support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -1,411 +1,149 @@
 package com.gu.support.workers.lambdas
 
-import cats.implicits._
 import com.amazonaws.services.lambda.runtime.Context
 import com.gu.config.Configuration
-import com.gu.monitoring.SafeLogger
-import com.gu.salesforce.Salesforce.SfContactId
 import com.gu.services.{ServiceProvider, Services}
-import com.gu.support.catalog
-import com.gu.support.catalog._
-import com.gu.support.config.{TouchPointEnvironments, ZuoraConfig}
-import com.gu.support.promotions.PromotionService
-import com.gu.support.redemption._
+import com.gu.support.config.TouchPointEnvironments
 import com.gu.support.redemption.corporate._
-import com.gu.support.redemption.gifting.GiftCodeValidator
-import com.gu.support.redemption.gifting.generator.GiftCodeGeneratorService
-import com.gu.support.redemptions.RedemptionData
-import com.gu.support.workers.GiftRecipient.DigitalSubscriptionGiftRecipient
 import com.gu.support.workers._
-import com.gu.support.workers.lambdas.DigitalSubscriptionGiftRedemption.{maybeDigitalSubscriptionGiftRedemption, redeemGift}
-import com.gu.support.workers.lambdas.NextState.{DigitalSubscriptionGiftPurchaseDetails, PaymentMethodWithSchedule}
-import com.gu.support.workers.states.SendThankYouEmailState._
-import com.gu.support.workers.states.{CreateZuoraSubscriptionState, SendAcquisitionEventState}
-import com.gu.support.zuora.api.ReaderType.Gift
-import com.gu.support.zuora.api._
-import com.gu.support.zuora.api.response.{Subscription, ZuoraAccountNumber, ZuoraSubscriptionNumber, ZuoraSuccessOrFailureResponse}
-import com.gu.support.zuora.domain.DomainSubscription
+import com.gu.support.workers.states.CreateZuoraSubscriptionState._
+import com.gu.support.workers.states.{CreateZuoraSubscriptionWrapperState, SendAcquisitionEventState, SendThankYouEmailState}
+import com.gu.zuora.ZuoraSubscriptionCreator
+import com.gu.zuora.productHandlers._
 import com.gu.zuora.subscriptionBuilders._
-import com.gu.zuora.{ZuoraGiftService, ZuoraSubscribeService}
-import org.joda.time.{DateTime, DateTimeZone, Days, LocalDate}
+import org.joda.time.{DateTime, DateTimeZone}
 
 import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
-import scala.util.{Failure, Success, Try}
 
 class CreateZuoraSubscription(servicesProvider: ServiceProvider = ServiceProvider)
-  extends ServicesHandler[CreateZuoraSubscriptionState, SendAcquisitionEventState](servicesProvider) {
+  extends ServicesHandler[CreateZuoraSubscriptionWrapperState, SendAcquisitionEventState](servicesProvider) {
```

Comment:
this file has changed (simplified!) so significantly that it's worth just viewing it alone rather than as a diff to ensure that it makes sense in its own right.